### PR TITLE
fixes #3206 - prevent any user including admin from creating lookup_value with fdqn= or hostgroup= if host/hostgroup does not exist

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/lookup_value.rb [hunk 1]
```diff
@@ -55,6 +56,20 @@ def validate_list
     errors.add(:value, _("%{value} is not one of %{rules}") % { :value => value, :rules => lookup_key.validator_rule }) and return false unless lookup_key.validator_rule.split(LookupKey::KEY_DELM).map(&:strip).include?(value)
   end
 
+  def ensure_fqdn_exists
+    md = match.match(/^fqdn=(.*)/)
+    return true unless md
+    return true if Host.find_by_name(md[1]) || host_or_hostgroup.try(:new_record?)
```
Comment (domcleal) [inline app/models/lookup_value.rb]: I think this might need to be .unscoped, same for host groups.

Comment (isratrade) [inline app/models/lookup_value.rb]: good call, fixed

File: test/unit/host_test.rb [hunk 2]
```diff
@@ -253,7 +253,7 @@ def setup_filtered_user
     as_admin do
       @one.roles = [Role.find_by_name("Viewer")]
     end
-    assert !@host.update_attributes(:name => "blahblahblah")
+    assert !@host.update_attributes(:comment => "blahblahblah")
```
Comment (isratrade) [inline test/unit/host_test.rb:256]: updating `name` was causing test failures 
Match fqdn=blahblahblah.mydomain.net does not match an existing host

**Comment by domcleal** [comment]
Thanks @isratrade, merged as bd48d5c93.

**Pull request closed without merging**